pygitops/operations.py

```diff
@@ -152,42 +153,38 @@ def feature_branch(repo: Repo, branch_name: str) -> Iterator[None]:
             )
 
 
-def get_updated_repo(
-    repo_url: str, clone_dir: PathOrStr, repo_name: str, **kwargs
-) -> Repo:
+def get_updated_repo(repo_url: str, clone_dir: PathOrStr, **kwargs) -> Repo:
     """
     Clone the default branch of the target repository, returning a repo object.
 
     If repo is already present, we will pull in the latest changes to the default branch of the repo.
 
     :param repo_url: URL of the Github repository to be cloned.
     :param clone_dir: The root directory where repositories are cloned.
-    :param repo_name: The name of the repository to be cloned.
     :raises PyGitOpsError: There was an error cloning the repository.
     """
     # make sure it's actually a Path if our user passed a str
     clone_dir = Path(clone_dir)
 
-    dest_repo_path = clone_dir / repo_name
-    git_lockfile_path = _get_lockfile_path(repo_name)
+    git_lockfile_path = _get_lockfile_path(str(clone_dir))
 
     # Lock the following operation such that only one process will attempt to clone the repo at a time.
     with FileLock(str(git_lockfile_path)):
         try:
             # if the repo already exists, don't clone it
-            if dest_repo_path.is_dir():
-                repo = Repo(dest_repo_path)
+            if _is_git_repo(clone_dir):
+                repo = Repo(clone_dir)
                 # pull down latest changes from `branch` if provided in kwargs, deferring to repo default branch
                 branch = kwargs.get("branch") or get_default_branch(repo)
                 _checkout_pull_branch(repo, branch)
                 return repo
 
-            return Repo.clone_from(repo_url, dest_repo_path, **kwargs)
+            return Repo.clone_from(repo_url, clone_dir, **kwargs)
```

Comment:
nice! (I read up on the `git clone` and `gitpython` documentation to confirm that this is what we want, I wasnt 100% sure)

this looks like what we want

Reply:
Ah good! Thanks for confirming!